src/server/pkg/deploy/assets/assets.go

```diff
@@ -579,6 +579,47 @@ func EtcdStatefulSet(opts *AssetOpts, diskSpace int) interface{} {
 		etcdCmd[i] = fmt.Sprintf("\"%s\"", str) // quote all arguments, for shell
 	}
 
+	var pvcTemplates []interface{}
+	switch backend {
+	case googleBackend, amazonBackend:
+		pvcTemplates = []interface{}{
+			map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"name":   etcdVolumeClaimName,
+					"labels": labels(etcdName),
+					"annotations": map[string]string{
+						"volume.beta.kubernetes.io/storage-class": etcdStorageClassName,
+					},
+				},
+				"spec": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"requests": map[string]interface{}{
+							"storage": resource.MustParse(fmt.Sprintf("%vGi", diskSpace)),
+						},
+					},
+					"accessModes": []string{"ReadWriteOnce"},
+				},
+			},
+		}
+	default:
+		pvcTemplates = []interface{}{
```

Comment:
Would it make sense to factor this out? Like:
```
pvcTemplates = []interface{}{
  ...
}
```
then
```
if backend == googleBackend || backend == amazonBackend {
  pvcTemplates[0]["metadata"]["annotations"] = map[string]string{
    "volume.beta.kubernetes..": etcdStorageClassName
  }
}
```

Also, if I understand this code, all it does is make sure that the `volume.beta.kubernetes...` label isn't set if you're not using those two backends, is that correct? Is the idea that you'll get dynamic provisioning in azure as long as you don't set that label?